### ISSUE TITLE
Signature help session is not triggered after typing comma in target-typed new expression inside return statement

### ISSUE THREAD

**Issue by Peter-Juhasz** [dotnet/roslyn#76499]
**Version Used**: VS 17.13 P2.1

**Steps to Reproduce**:

https://github.com/user-attachments/assets/4bc1add9-a4e4-446f-b2f3-38e08c79c4e4

```cs
static A Do()
{
    return new("ASD"$$);
}

public record class A(string Id, int Count);
```

**Expected Behavior**:
Signature help session starts.

**Actual Behavior**:
Signature help session doesn't start.

**Comment by CyrusNajmabadi** [comment]
Debugged through.  This is because the compiler is giving back an error type symbol when we ask what the type of this expression is (including the converted type of it).

While tehre's def a problem in that a specific constructor cannot be found.  We should still be able to figure out the type here.  Assigning to compiler to take a look (note: i think we've heard abotu this in teh past.  not sure if there's a bug to dupe this against).

**Comment by davidwengier** [comment]
Ran in to this today, and also noticed completion doesn't offer parameter names making it very annoying to deal with constructors with lots of optional parameters. Probably the same root cause, but thought I'd comment because the experience is particularly painful in this scenario.

**Comment by RikkiGibson** [comment]
@CyrusNajmabadi could you please clarify which API is not behaving as expected? The following unit test passes, so I am thinking that the issue is not with `GetTypeInfo`?

```cs
[Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76499")]
public void Repro_76499_1()
{
    var source = """
        static A Do()
        {
            return new("ASD");
        }

        public record class A(string Id, int Count);
        """;

    var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, parseOptions: TestOptions.RegularPreview);
    comp.VerifyEmitDiagnostics(
        // (1,10): warning CS8321: The local function 'Do' is declared but never used
        // static A Do()
        Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "Do").WithArguments("Do").WithLocation(1, 10),
        // (3,12): error CS7036: There is no argument given that corresponds to the required parameter 'Count' of 'A.A(string, int)'
        //     return new("ASD");
        Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, @"new(""ASD"")").WithArguments("Count", "A.A(string, int)").WithLocation(3, 12));

    var tree = comp.SyntaxTrees[0];
    var model = comp.GetSemanticModel(tree);
    var node = tree.GetRoot().DescendantNodes().OfType<ImplicitObjectCreationExpressionSyntax>().Single();
    var typeInfo = model.GetTypeInfo(node);
    Assert.Equal("A", typeInfo.Type.ToTestDisplayString());
}
```

**Comment by CyrusNajmabadi** [comment]
Try with things like an extra comma in the argument list. I believe we also call GetMemberGroup since we want to know all the overloads to offer.
